docs/src/threading-java.md

```diff
@@ -55,4 +55,25 @@ public class PlaywrightThread extends Thread {
     }
   }
 }
+```
+
+## Synchronous API and event dispatching
+
+In the synchronous Playwright API all events are dispatched only when Playwright is running its message loop.
+This happens automatically when you call any of the API methods and doesn't happen if there are no active
+Playwright calls on the stack. If you need to wait for an event the best way to do this is via one of the
+`waitFor*` methods.
+
+### Page.waitForTimeout() vs. Thread.sleep()
+
+One consequence of the synchronous API is that if you for whatever reason call `Thread.sleep()` no events will
+be fired while the thread is sleeping. If you want events from the browser to be dispatched while the program
+execution is paused use [`method: Page.waitForTimeout`] or [`method: Frame.waitForTimeout`]:
```

Comment:
We have the same problem in Python and probably also in .NET: https://playwright.dev/python/docs/intro#timesleep-leads-to-outdated-state

Reply:
Do you suggest moving this to the intro section? I'd rather keep that more concise and focused on the initial experience but perhaps we should restructure it.